catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/history/HistorianTest.java

```diff
@@ -0,0 +1,462 @@
+
+        storageProvider = new InMemoryStorageProvider();
+        historian.setStorageProviders(Collections.singletonList(storageProvider));
+
+        historian.setFilterBuilder(new GeotoolsFilterBuilder());
+
+        historian.setMetacardTypes(Collections.singletonList(BasicTypes.BASIC_METACARD));
+
+        Security security = mock(Security.class);
+        Subject subject = mock(MockSubject.class);
+        when(subject.execute(any(Callable.class))).thenCallRealMethod();
+        when(security.getSystemSubject()).thenReturn(subject);
+        historian.setSecurity(security);
+    }
+
+    @Test
+    public void testDoSkip() throws SourceUnavailableException, IngestException {
+        historian.setHistoryEnabled(false);
+        historian.version(mock(UpdateResponse.class));
+
+        verifyZeroInteractions(catalogProvider);
+    }
+
+    @Test
+    public void testSetSkipFlag() throws SourceUnavailableException, IngestException {
+        Map<String, Serializable> properties = new HashMap<>();
+        UpdateResponse updateResponse = createUpdateResponse(properties);
+
+        historian.version(updateResponse);
+
+        assertThat(properties, hasEntry(MetacardVersion.SKIP_VERSIONING, true));
+    }
+
+    @Test
+    public void testSkipProperty() throws SourceUnavailableException, IngestException {
+        Map<String, Serializable> properties = new HashMap<>();
+        properties.put(MetacardVersion.SKIP_VERSIONING, true);
+
+        UpdateResponse updateResponse = createUpdateResponse(properties);
+
+        historian.version(updateResponse);
+        // Called once for skip check
+        verify(updateResponse).getProperties();
+    }
+
+    @Test
+    public void testUpdateResponse() throws Exception {
+        UpdateResponse updateResponse = createUpdateResponse(null);
+        List<Update> updateList = createUpdatedMetacardList();
+        when(updateResponse.getUpdatedMetacards()).thenReturn(updateList);
+
+        historian.version(updateResponse);
+        ArgumentCaptor<CreateRequest> createRequest = ArgumentCaptor.forClass(CreateRequest.class);
+        verify(catalogProvider).create(createRequest.capture());
+
+        Metacard versionedMetacard = createRequest.getValue()
+                .getMetacards()
+                .get(0);
+        assertThat(versionedMetacard.getAttribute(MetacardVersion.VERSION_OF_ID)
+                .getValue(), equalTo(METACARD_ID));
+    }
+
+    @Test
+    public void testUpdateStorageResponseSkip()
+            throws UnsupportedQueryException, SourceUnavailableException, IngestException {
+        UpdateStorageRequest storageRequest = mock(UpdateStorageRequest.class);
+        UpdateStorageResponse storageResponse = mock(UpdateStorageResponse.class);
+        UpdateResponse updateResponse = mock(UpdateResponse.class);
+
+        historian.version(storageRequest, storageResponse, updateResponse);
+
+        assertThat(storageProvider.storageMap.size(), equalTo(0));
```

Comment:
Can't verify zero interactions with the storage provider (not a mock), but added a check for that catalog provider.